crates/swc_ecma_utils/src/lib.rs

```diff
@@ -2165,6 +2165,14 @@ where
 {
     noop_visit_type!();
 
+    fn visit_assign_pat_prop(&mut self, node: &AssignPatProp, _: &dyn Node) {
```

Comment:
This does not fix the issue completely. I will open a PR.

Reply:
Oh, thanks!
I thought it was enough.

Reply:
I think we need to handle `const [ test ] = asdf;`. I'm just writing a test to confirm it.

Reply:
[playground link](https://play.swc.rs/?version=1.2.106&code=H4sIAAAAAAAAA%2BNKzs8rLlGIVkjLz1eIVbBVqK4FALEg6qoTAAAA&config=H4sIAAAAAAAAA0WNSwrDMAxE76J1ts0id%2BghjKsEF%2F%2FQKBBjfPfawSU7aebpqdIXlrZK2QhYxoQS1Vy0kZbMsOKy0kKKHu3Gg1tfjBysHWG8eudTAs92oeCi28sw2RSyMPBUJh7%2BT7YuCulzjqDez27hSu1xzDuH9wRVTm4%2FOkwQUrUAAAA%3D)


You can change to AST mode to see how array patterns are consturcted.


```json
{
  "type": "Module",
  "span": {
    "start": 2633,
    "end": 2651,
    "ctxt": 0
  },
  "body": [
    {
      "type": "VariableDeclaration",
      "span": {
        "start": 2633,
        "end": 2651,
        "ctxt": 0
      },
      "kind": "const",
      "declare": false,
      "declarations": [
        {
          "type": "VariableDeclarator",
          "span": {
            "start": 2639,
            "end": 2651,
            "ctxt": 0
          },
          "id": {
            "type": "ArrayPattern",
            "span": {
              "start": 2639,
              "end": 2646,
              "ctxt": 0
            },
            "elements": [
              {
                "type": "Identifier",
                "span": {
                  "start": 2641,
                  "end": 2644,
                  "ctxt": 0
                },
                "value": "foo",
                "optional": false,
                "typeAnnotation": null
              }
            ],
            "optional": false,
            "typeAnnotation": null
          },
          "init": {
            "type": "ObjectExpression",
            "span": {
              "start": 2649,
              "end": 2651,
              "ctxt": 0
            },
            "properties": []
          },
          "definite": false
        }
      ]
    }
  ],
  "interpreter": null
}
```

Reply:
Thanks @kdy1! It is actually working. My bad! I misread the code—ArrayPatterns have elements that are `Pat` and so it will visit those identifiers.

I'm going to submit my tests in a PR though 🙂

Reply:
Thanks!